### PR TITLE
Add mapper comptibility for Que 1.x

### PR DESCRIPTION
**Pull request by mrrooijen**

### CODE REVIEW

**Comment by defkode** [comment]
I am not insisting on AR here, I get your point about users with sequel, let's keep it simple without mapper option. What is more important in my opinion here is the query itself:

New Que version should filter out finished & expired_jobs, old version doesn't have it:

```SQL
SELECT COUNT(*) FROM que_jobs WHERE finished_at IS NULL AND expired_at IS NULL AND run_at <= NOW() AND queue = 'mailers'
```

vs 

```SQL
SELECT COUNT(*) FROM que_jobs WHERE  run_a" <= NOW() AND queue = 'mailers'
```

**Comment by mrrooijen** [comment]
Thanks, that looks good! I've updated the PR (see: https://github.com/hirefire/hirefire-resource/pull/59/files)

Thoughts?

**Comment by mrrooijen** [comment]
Query example:

Que 0.x

```
all
"SELECT count(*) AS total FROM que_jobs WHERE run_at < NOW()"
worker
"SELECT count(*) AS total FROM que_jobs WHERE run_at < NOW() AND queue IN ('worker')"
worker, mailer
"SELECT count(*) AS total FROM que_jobs WHERE run_at < NOW() AND queue IN ('worker','mailer')"
```

Que 1.x

```
all
"SELECT COUNT(*) FROM que_jobs WHERE finished_at IS NULL AND expired_at IS NULL AND run_at <= NOW()"
worker
"SELECT COUNT(*) FROM que_jobs WHERE finished_at IS NULL AND expired_at IS NULL AND run_at <= NOW() AND queue IN ('worker')"
worker, mailer
"SELECT COUNT(*) FROM que_jobs WHERE finished_at IS NULL AND expired_at IS NULL AND run_at <= NOW() AND queue IN ('worker','mailer')"
```

**Comment by defkode** [comment]
elegant! Thank you

File: lib/hirefire/macro/que.rb [hunk 1]
```diff
@@ -20,17 +16,40 @@ module Que
...
+      end
+
+      QUE_V0_QUERY = query_const(<<-QUERY)
+        SELECT count(*) AS total
+        FROM que_jobs
+        WHERE run_at < NOW()
+      QUERY
+
+      QUE_V1_QUERY = query_const(<<-QUERY)
+        SELECT COUNT(*)
```
Comment (defkode) [inline lib/hirefire/macro/que.rb:47]:
```suggestion
        SELECT COUNT(*) AS total
```

File: lib/hirefire/macro/que.rb [hunk 2]
```diff
@@ -20,17 +16,40 @@ module Que
...
+      def names(queues)
+        queues.map { |queue| "'#{queue}'" }.join(",")
+      end
+
+      def query_const(query)
+        query.gsub(/\s+/, " ").strip.freeze
+      end
+
+      QUE_V0_QUERY = query_const(<<-QUERY)
+        SELECT count(*) AS total
```
Comment (defkode) [inline lib/hirefire/macro/que.rb:41]: use same case as query below

```suggestion
        SELECT COUNT(*) AS total
```

**Comment by mrrooijen** [comment]
Thanks for the review! I've applied the changes and released v0.9.0.

**Pull request merged**